drivers/api/irq.c

```diff
@@ -92,9 +92,10 @@ int32_t irq_register_callback(struct irq_ctrl_desc *desc, uint32_t irq_id,
  * @param callback - Callback descriptor - platform specific type.
  * @return SUCCESS in case of success, FAILURE otherwise.
  */
-int32_t irq_unregister(struct irq_ctrl_desc *desc, uint32_t irq_id)
+int32_t irq_unregister_callback(struct irq_ctrl_desc *desc, uint32_t irq_id,
+				void *callback)
```

Comment:
As stated in the conversation tab, in my opinion one slot of the interrupt vector should only accept one interrupt handler, or in this case one callback. If a peripheral only has one line, but multiple sources, those should be determined and serviced in the same callback.

Reply:
On stm32, the callbacks are `void callback(void)` so they need to be associated beforehand with a certain event.
Some other platforms may have a parameter or two or three to the callback, from which the code within the callback may tell which event triggered the callback.

Point is, the current implementation tries to be flexible enough to accommodate all these. And this is achieved using the 3rd parameter of irq_register_callback() being now a void*.